### PR TITLE
#6077: Multiple selection on Map

### PR DESCRIPTION
**Pull request by AGMETEOR**
## Description

Allow Multiple selection and deselection when using Geometry filter

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [x] Other... Please describe: Enhancement

## Issue

**What is the current behavior?**

#6077

**What is the new behavior?**

Screen-Recording-2020-10-29-at-14 20 34

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
On Mac OS, use the command key instead of Ctrl

### CODE REVIEW

File: web/client/epics/featuregrid.js [hunk 1]
```diff
@@ -290,10 +291,21 @@ module.exports = {
             .map(updateFilterFunc(store))
     ),
     /**
-     * Enables the Geometry filter when entering edit mode in feature grid.
-     * No filter value should have been set otherwise nothing is enabled.
-     * @memberof epics.featuregrid
+     * Resets multiselect and ctrl, metaKey modifiers in case they were true during map clicks
+     * This happens when filter is disabled
+     * @memberof epics.featuregird
```
Comment (MV88) [inline web/client/epics/featuregrid.js:296]:
```suggestion
     * @memberof epics.featuregrid
```

File: web/client/reducers/mapInfo.js [hunk 2]
```diff
@@ -288,6 +289,12 @@ function mapInfo(state = initState, action) {
             filterNameList: action.filterNameList || null
         });
     }
+    case UPDATE_MAPINFO_STATE_WITH_CLICK_POINT: {
```
Comment (MV88) [inline web/client/reducers/mapInfo.js:292]: could you not reuse FEATURE_INFO_CLICK ?

**Review by MV88 — CHANGES_REQUESTED** [review]
Hi, seems good, but the only problem i've found is that the numbers are not updated with the current rows.

it is always shown **1 item of 1**

File: web/client/reducers/mapInfo.js (hunk 2 above)
Comment (offtherailz) [reply web/client/reducers/mapInfo.js:292]: More precisely I think you should intercept the CLICK_ON_MAP action as well as the actual filter does. Click on map action should contain all the info, included the modifiers, so you don't need anything else, i think.

Comment (offtherailz) [reply web/client/reducers/mapInfo.js:292]: Map info is dedicated to feature info tool. In this case the click on map is handled by the other tool, so the mapinfo actions/apics should not be included at all

**Comment by AGMETEOR** [comment]
@MV88 about this [comment](https://github.com/geosolutions-it/MapStore2/pull/6121#pullrequestreview-519826589) I can't seem to see the numbers info anymore. Maybe something was changed in the FeatureEditor code?

File: web/client/reducers/mapInfo.js (hunk 2 above)
Comment (MV88) [reply web/client/reducers/mapInfo.js]: > @MV88 about this [comment](https://github.com/geosolutions-it/MapStore2/pull/6121#pullrequestreview-519826589) I can't seem to see the numbers info anymore. Maybe something was changed in the FeatureEditor code?

i mean the text at the bottom left when you have multiple rows selected
image

Comment (AGMETEOR) [reply web/client/reducers/mapInfo.js]: Alright. I had a small screen size with console open so I wasn't seeing it. I have fixed the count issue.

**Comment by tdipisa** [comment]
@AGMETEOR is it ready for a new review?

**Review by MV88 — CHANGES_REQUESTED** [review]
I've found three points that needs to be addressed:

1) improve tooltip for the selection because the user cannot be aware of using the CTRL for multiple selection

2) when you have selected a feature, then you enter query panel and exit, without searching, you 
 2.1) lose the highlight style on the selected features see gif below
 2.2) only get one row instead of the previous selected ones  (3 in the gif)

multiple selection

**Comment by AGMETEOR** [comment]
@MV88 requested a new review on this. Fixed above issues in your latest comment

**Review by MV88 — CHANGES_REQUESTED** [review]
Other points i have found:

1) when you have selected something with the clicks and you open the query panel they should be hidden (temporarily)

then 

1.1)  when clicking on the left arrow, going back they should be shown again

1.2) when instead i apply a search they should be cleared and the filter coming from query panel be applied.

Another minor thing, the message passes from **1 items of 1** to 2 items, 3 items. 
can't you have **1 item** as initial text?

File: web/client/translations/data.it-IT.json [hunk 3]
```diff
@@ -1541,7 +1541,7 @@
             "deselectall": "Deseleziona tutti",
             "backtosearch": "Ritorna alla ricerca",
             "resultInfo": "{total, plural, =0 {Nessun elemento} =1 {{total} di {total}} other {{start}-{end} di {total}}}",
-            "resultInfoVirtual": "{total, plural, =0 {Nessun elemento} =1 {{total} di {total}} other {{total} Elementi}}",
+            "resultInfoVirtual": "{total, plural, =0 {Nessun elemento} =1 {{total} Elementi} other {{total} Elementi}}",
```
Comment (MV88) [inline web/client/translations/data.it-IT.json:1544]:
```suggestion
            "resultInfoVirtual": "{total, plural, =0 {Nessun elemento} =1 {{total} Elemento} other {{total} Elementi}}",
```

**Review by MV88 — CHANGES_REQUESTED** [review]
can you check it works properly_? i see that it loses the number of selected features when closing the query panel

**Review by MV88 — APPROVED** [review]
LGTM

**Pull request merged**